### PR TITLE
Fix microversion while resizing resource_openstack_containerinfra_nodegroup_v1 to 0

### LINKED ISSUE #1736: Wrong client microversion while resizing nodegroup to 0
### Terraform Version
```
Terraform v1.7.5
on linux_amd64
+ provider registry.terraform.io/terraform-provider-openstack/openstack v2.0.0
```

### Affected Resource(s)
- resource_openstack_containerinfra_nodegroup_v1

### Terraform Configuration Files
```hcl
resource "openstack_containerinfra_nodegroup_v1" "nodegroup_ng" {
  name                = "ng"
  cluster_id          = openstack_containerinfra_cluster_v1.cluster[count.index].id
  docker_volume_size  = 10
  min_node_count      = 0
  max_node_count      = 5
  node_count          = 0
  flavor_id           = flavorID
  count = length(openstack_containerinfra_cluster_v1.cluster[*].id)  
}
```

### Output
```
Terraform will perform the following actions:

  # openstack_containerinfra_nodegroup_v1.nodegroup_ng[0] will be updated in-place
  ~ resource "openstack_containerinfra_nodegroup_v1" "nodegroup_ng" {
        id                 = "0259858b-077e-4cb0-8b22-aff8447ba367/17c18545-4735-4253-bbe0-f7ba93d858fc"
        name               = "ng"
      ~ node_count         = 1 -> 0
        # (13 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions in workspace "test"?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

openstack_containerinfra_nodegroup_v1.nodegroup_ng[0]: Modifying... [id=0259858b-077e-4cb0-8b22-aff8447ba367/17c18545-4735-4253-bbe0-f7ba93d858fc]
╷
│ Error: Error resizing openstack_containerinfra_nodegroup_v1 0259858b-077e-4cb0-8b22-aff8447ba367/17c18545-4735-4253-bbe0-f7ba93d858fc: Bad request with: [POST https://openstack:13511/v1/clusters/0259858b-077e-4cb0-8b22-aff8447ba367/actions/resize], error message: {"errors": [{"request_id": "", "code": "client", "status": 400, "title": "Resizing a nodegroup to zero is not supported in the provided microversion", "detail": "Resizing a nodegroup to zero is not supported in the provided microversion.", "links": []}]}
│ 
│   with openstack_containerinfra_nodegroup_v1.nodegroup_ng[0],
│   on nodegroup.tf line 13, in resource "openstack_containerinfra_nodegroup_v1" "nodegroup_ng":
│   13: resource "openstack_containerinfra_nodegroup_v1" "nodegroup_ng" {
│ 
```

### Expected Behavior
Nodegroup should scale down to 0.

### Actual Behavior
Provider sends wrong client microversion to Openstack API - 1.9 instead of 1.10 required to scale down nodegroup to 0

### Steps to Reproduce
1. Create Magnum cluster
2. Create additional Nodegroup with 1 node
3. Change node_count parameter for additional Nodegroup to 0
4. Run `terraform apply`

### PR DESCRIPTION
**Pull request by pawcykca**
This PR fixes error with wrong client microversion (1.9 instead of 1.10) while resizing additional nodegorup to 0.

Fixes #1736

### CODE REVIEW

File: openstack/resource_openstack_containerinfra_nodegroup_v1.go [hunk 1]
```diff
@@ -302,28 +302,33 @@ func resourceContainerInfraNodeGroupV1Update(ctx context.Context, d *schema.Reso
...
-			Timeout:      d.Timeout(schema.TimeoutUpdate),
-			Delay:        1 * time.Minute,
-			PollInterval: 20 * time.Second,
-		}
-		_, err = stateConf.WaitForStateContext(ctx)
-		if err != nil {
-			return diag.Errorf(
-				"Error waiting for openstack_containerinfra_node_group_v1 %s to become updated: %s", d.Id(), err)
+		nodeCount := d.Get("node_count").(int)
+		if nodeCount >= 0 {
```
Comment (kayrus) [inline openstack/resource_openstack_containerinfra_nodegroup_v1.go:306]: Can you add a positive check inside the `"node_count":` argument definition? E.g. `ValidateFunc: validation.IntAtLeast(0),`

Comment (pawcykca) [reply openstack/resource_openstack_containerinfra_nodegroup_v1.go]: Good suggestion, changed.

**Review by kayrus — APPROVED** [review]
LGTM, thanks!

**Review by nikParasyr — APPROVED** [review]
LGTM

**Pull request merged**